README.md

```diff
@@ -7,15 +7,198 @@
+selected slide index is unnecessarily complex. E.g. if we add a photo collection,
+we need to add the `currentSlide` property to the store as well.
+
+Simply distinguishing between "component UI" state and global application state
+radically simplified the architecture in the above case, since component state
+can by definition not exist without a matching component (and vice versa).
+
+2. Simplified Testing
+
+Testing React component is extremely easy compared to other frameworks, such as
+Angular. Reason being that React components are by definition declarative, while
+Angular heavily relies on imperative APIs.
+
+Using stores doesn't necessarily break this abstraction, but it makes it much
+harder to properly test all the possible states that a component can be in,
+since a store might contain state that isn't directly consumed by the component
+to be tested.
+
+But more importantly, we now need to manage a store during testing. This means
+we need to make sure we **always** restore it to its previous state in an
+`beforeEach`, otherwise this can lead to hard to debug failed tests.
+
+Instead of adding a `beforeEach`, we can also mock it out, which eliminates the
+need to reset the store.
+
+3. Reusing the component is possible
+
+While we focused implementing our own set of stores, some people prefer to use
+Redux, Rx, mobx or some other library for managing state and implementing
+unidirectional data flow.
+
+By storing state in an external store, we implicitly couple the component to
+whatever architecture we chose for our main application. If we're implementing
+an accordion component using Flux, it means everyone using our component will
+have to use Flux in order to interact with it (even though it might be hidden
+though the public API of the component).
+
+Hence using component state (and props) instead of stores is the preferred way
+when creating reusable components.
+
+## Connecting components using lifecycle methods
+
+So far we oftentimes "connected" components to stores by adding an event
+listener on the store in the component's lifecycle methods:
+
+```js
+class MyCounter extends React.Component {
+	componentDidMount() {
+		this.removeListener = store.addListener(count => {
+			this.setState({ count });
+		});
+	}
+	componentWillUnmount() {
+		this.removeListener();
+	}
+}
+```
+
+In `componentDidMount` as add an event listener that updates the component's
+state based on the stores state, thus triggering a re-rendering.
+
+In `componentWillUnmount` we remove the event listener. Typically `addListener`
+returns a function that — when invoked — removes the event listener from the
+store. In other cases people might subclass the `EventEmitter` class directly
+in order to implement a store, hence there are cases in which we actually have
+to call `removeListener` on the store itself.
+
+`componentDidMount` and `componentWillUnmount` are so-called lifecycle methods,
```

Comment:
We can remove this paragraph, students will have already learned about lifecycle methods